internal/k8s/controller.go

```diff
@@ -1794,37 +1728,38 @@ func (lbc *LoadBalancerController) createIngressEx(ing *networking.Ingress, vali
 		ValidMinionPaths: validMinionPaths,
 	}
 
-	ingEx.TLSSecrets = make(map[string]*api_v1.Secret)
+	ingEx.Secrets = make(map[string]*configs.SecretReference)
+
 	for _, tls := range ing.Spec.TLS {
 		secretName := tls.SecretName
 		secretKey := ing.Namespace + "/" + secretName
-		secret, err := lbc.getAndValidateSecret(secretKey)
+
+		secretType, secretPath, err := lbc.secretStore.GetSecret(secretKey)
 		if err != nil {
 			glog.Warningf("Error trying to get the secret %v for Ingress %v: %v", secretName, ing.Name, err)
-			continue
 		}
-		ingEx.TLSSecrets[secretName] = secret
+
+		ingEx.Secrets[secretName] = &configs.SecretReference{
+			Type:  secretType,
+			Path:  secretPath,
+			Error: err,
+		}
 	}
 
 	if lbc.isNginxPlus {
 		if jwtKey, exists := ingEx.Ingress.Annotations[configs.JWTKeyAnnotation]; exists {
 			secretName := jwtKey
+			secretKey := ing.Namespace + "/" + secretName
 
-			secret, err := lbc.client.CoreV1().Secrets(ing.Namespace).Get(context.TODO(), secretName, meta_v1.GetOptions{})
+			secretType, secretPath, err := lbc.secretStore.GetSecret(secretKey)
 			if err != nil {
-				glog.Warningf("Error retrieving secret %v for Ingress %v: %v", secretName, ing.Name, err)
-				secret = nil
-			} else {
-				err = ValidateJWKSecret(secret)
-				if err != nil {
-					glog.Warningf("Error validating secret %v for Ingress %v: %v", secretName, ing.Name, err)
-					secret = nil
-				}
+				glog.Warningf("Error trying to get the secret %v for Ingress %v: %v", secretName, ing.Name, err)
```

Comment:
Should we add some context about what namespace to at least one of the logs about the resources?
```suggestion
				glog.Warningf("Error trying to get the secret %v for Ingress %v/%v: %v", secretKey, ing.Namespace, ing.Name, err)
```